R/pipeline.R

```diff
@@ -152,7 +152,7 @@ orchestrate_pipeline <- function(config_path,
 #'
 #' @rdname pipeline
 #' @export
-execute_model_logic <- function(config, output_dir) {
+execute_model_logic <- function(config, output_dir, blob_storage_container) {
```

Comment:
Why is the `blob_storage_container` being added to `execute_model_logic`? It wouldn't appear to be related to making metadata value atomic to me. What about if the `execute_model_logic` function is to be run locally without any `blob_storage_container`? At the moment `blob_storage_container` is a required argument, perhaps set to default to `""` or `NULL`?

Reply:
Yes, this is something I struggled with a bit. I think ideally, the metadata would be returned from `execute_model_logic()` as a value, and `orchestrate_pipeline()` would be the function that writes the values to disk. Thinking through what this would require:
- `write_model_outputs()` does not write the metadata. Alter it to **return** the `samples_path`, `summaries_path`, and `model_path` it used when writing the samples, summaries, and model.
  - Alter the tests that check the return value and metadata saving.
- `execute_model_logic()` accepts the returned paths from `write_model_outputs()`, and adds them to the `metadata` list. It then returns the metadata list. Also add model completion status to the metadata?
  - Alter the tests that check the return value.
- `orchestrate_pipeline()` accepts the metadata list from `execute_model_logic()`. It adds the blob storage container to the metadata. It writes the metadata to file, using the `dirname()` of the model path.
  - Alter tests that check the output of `orchestrate_pipeline()`. Add tests for checking that metadata has been saved to disk.

This is ... a bit bigger than I had originally intended this PR to be. Should it be done in a separate PR? Or just bite the bullet and do it here?

Reply:
Would be happy for this to be in a new PR! Thanks.

Reply:
Moved to #122